app/services/twilio/incoming_message_service.rb

```diff
@@ -126,4 +122,26 @@ def attach_files
 
     @message.save!
   end
+
+  def download_attachment_file
+    download_with_auth
+  rescue Down::Error => e
+    handle_download_attachment_error(e)
+  end
+
+  def download_with_auth
+    Down.download(
+      params[:MediaUrl0],
+      # https://support.twilio.com/hc/en-us/articles/223183748-Protect-Media-Access-with-HTTP-Basic-Authentication-for-Programmable-Messaging
+      http_basic_authentication: [twilio_channel.account_sid, twilio_channel.auth_token || twilio_channel.api_key_sid]
+    )
+  end
+
+  def handle_download_attachment_error(error)
+    if error.message.include?('401 Unauthorized')
```

Comment:
Is there a way we can get an HTTP error code instead of checking against a string?